### ISSUE TITLE
Debug nonce too low error

### ISSUE THREAD

**Issue by salieflewis** [1ifeworld/river#338]
A few recent transactions have been failing due to low nonce errors. Here is the log from Vercel.

```
Error in relayPost:  TransactionExecutionError: Nonce provided for the transaction (655) is lower than the current nonce of the account.
Try increasing the nonce or find the latest nonce with `getTransactionCount`.
```

Now I'm tempted to throw the txn signer on the server too. Thirdweb engine? 

What do you think could be wrong with the current Lattice impl? What did we do differently?

cc: @0xTranqui

**Comment by 0xTranqui** [comment]
seeing this log doesnt necessarily mean the transaction didnt go thru, since there is built in retrying following a nonce reset. i think its supposed to be able to retry 3 times. if txns arent going thru it could def still be this, but potentially something else too

**Comment by 0xTranqui** [comment]
peep line 100 of here https://github.com/1ifeworld/river/blob/main/apps/site/lib/transactions/writeContract.ts#L100. whether or not these retries are actually happening (maybe code is getting used incorrectly elsewhere) i dont know. but they retry at least once confirmed sometimes lol

**Comment by 0xTranqui** [comment]
> A few recent transactions have been failing due to low nonce errors. Here is the log from Vercel.
> 
> ```
> Error in relayPost:  TransactionExecutionError: Nonce provided for the transaction (655) is lower than the current nonce of the account.
> Try increasing the nonce or find the latest nonce with `getTransactionCount`.
> ```
> 
> Now I'm tempted to throw the txn signer on the server too. Thirdweb engine?
> 
> What do you think could be wrong with the current Lattice impl? What did we do differently?
> 
> cc: @0xTranqui

oh and putting this in server could be rlly cool! maybe third web engine approach like this is actually it

**Comment by salieflewis** [comment]
> seeing this log doesnt necessarily mean the transaction didnt go thru, since there is built in retrying following a nonce reset. i think its supposed to be able to retry 3 times. if txns arent going thru it could def still be this, but potentially something else too

Understood, I didn't realize that. Probably makes sense to move this out of the console then. Is there a blockscout link to the `PostGateway` somewhere, that is probably the best way to confirm things are going through, right?

**Comment by salieflewis** [comment]
Bumping this @0xTranqui cause I want to be aware of this while testing cause then we can close this!

**Comment by 0xTranqui** [comment]
@salieflewis i actually feel like its nice to have in the logs so that we can see its alive. maybe we could edit the logs to be more descriptive so we can tell if its just a retry or things bugging out?

**Comment by 0xTranqui** [comment]
heres the blockscout btw https://explorerl2new-river-j5bpjduqfv.t.conduit.xyz/address/0x1B692589017d4b2276227b52cD3A2a1796cb2b86

**Comment by salieflewis** [comment]
> @salieflewis i actually feel like its nice to have in the logs so that we can see its alive. maybe we could edit the logs to be more descriptive so we can tell if its just a retry or things bugging out?

I think its totally fine in dev, but I wouldn't want to include basically any logs on the client in prod.

**Comment by 0xTranqui** [comment]
> > @salieflewis i actually feel like its nice to have in the logs so that we can see its alive. maybe we could edit the logs to be more descriptive so we can tell if its just a retry or things bugging out?
> 
> I think its totally fine in dev, but I wouldn't want to include basically any logs on the client in prod.

do they not come up in the server?

**Comment by salieflewis** [comment]
> > > @salieflewis i actually feel like its nice to have in the logs so that we can see its alive. maybe we could edit the logs to be more descriptive so we can tell if its just a retry or things bugging out?
> > 
> > 
> > I think its totally fine in dev, but I wouldn't want to include basically any logs on the client in prod.
> 
> do they not come up in the server?

I don't believe so.

**Comment by 0xTranqui** [comment]
<img width="924" alt="Screenshot 2024-01-19 at 5 57 36 PM" src="https://github.com/1ifeworld/river/assets/93691906/408e7fab-84aa-49d8-8420-fc8884574977">
isnt this specifically a server error?

**Comment by salieflewis** [comment]
Here is a [link](https://vercel.com/1ifeworld/river-site/logs?page=1&timeline=past30Minutes&startDate=1705719207113&endDate=1705721007113&selectedLogId=170572093188193188105500000&selectedLogTimestamp=1705720931881&forceEndDate=1705720931881) to the specific error I received.

I'm trying to confirm this on Blockscout but it keeps timing out without fetching the latest transactions.

**Comment by salieflewis** [comment]
> <img alt="Screenshot 2024-01-19 at 5 57 36 PM" width="924" src="https://private-user-images.githubusercontent.com/93691906/298219938-408e7fab-84aa-49d8-8420-fc8884574977.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU3MjEzMjMsIm5iZiI6MTcwNTcyMTAyMywicGF0aCI6Ii85MzY5MTkwNi8yOTgyMTk5MzgtNDA4ZTdmYWItODRhYS00OWQ4LTg0MjAtZmM4ODg0NTc0OTc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIwVDAzMjM0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1YmQyYjZiYzIwOTk1YzQ4ZDk2ZGYxYTFmMDdhOGMyNzE3ZmEyZTExNjUxMGI5ODkyZTUzZjBjZGM5YTY0ZDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GnNdSXlOpW9oScKXkHx169wPDdF0pqWNaiRGHVZh87I"> isnt this specifically a server error?

Yes, you're right. I was confused.

**Comment by salieflewis** [comment]
Resolved #370
